lnclient/lnd/lnd.go

```diff
@@ -53,9 +53,23 @@ func NewLNDService(ctx context.Context, eventPublisher events.EventPublisher, ln
 		logger.Logger.WithError(err).Error("Failed to create new LND client")
 		return nil, err
 	}
-	nodeInfo, err := fetchNodeInfo(ctx, lndClient)
+
+	var nodeInfo *lnclient.NodeInfo
+	maxRetries := 5
+	for i := range maxRetries - 1 {
+		nodeInfo, err = fetchNodeInfo(ctx, lndClient)
+		if err == nil {
+			break
+		}
+		logger.Logger.WithFields(logrus.Fields{
+			"iteration": i,
+		}).WithError(err).Error("Failed to connect to LND, retrying in 10s")
+		time.Sleep(10 * time.Second)
+	}
+
+	nodeInfo, err = fetchNodeInfo(ctx, lndClient)
```

Comment:
why fetch again?

Reply:
Ah, that was to not have the conditional (see [this](https://github.com/getAlby/hub/pull/1219#discussion_r2010140451) and first commit)

Reply:
I think you can just remove line 70, you already store the error and nodeInfo because the variables are declared outside the loop.

The only issue then is doing an unnecessary 10 second sleep on the last iteration but this is an edge case and means LND couldn't be connected to anyway. So I don't think it matters.

What do you think?

Reply:
This is how the logs look like: (they make sense, but we are not actually retrying on the 6th time)

![Screenshot 2025-03-25 at 12 49 43 PM](https://github.com/user-attachments/assets/8aeea757-bc36-47b6-9fca-04b8f85cee02)

Reply:
I think it's ok, not worth the code duplication or conditional